src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,476 @@
+import Checkbox from "./../Checkbox/Checkbox";
+import MultiSelectItemsCountButton from "./MultiSelectItemsCountButton";
+import TextInput from "../TextInput/TextInput";
 
 export interface MultiSelectItem {
   id: string;
   name: string;
+  isDisabled?: boolean;
   children?: MultiSelectItem[];
 }
-
-export const multiSelectWidthsArray = [
-  "default",
-  "fitContent",
-  "full",
-] as const;
+export const multiSelectWidthsArray = ["fitContent", "full"] as const;
 export type MultiSelectWidths = typeof multiSelectWidthsArray[number];
-
+export const multiSelectListOverflowArray = ["scroll", "expand"] as const;
+export type MultiSelectListOverflowTypes =
+  typeof multiSelectListOverflowArray[number];
 export interface SelectedItems {
   [name: string]: { items: string[] };
 }
-interface MultiSelectCommonProps {
-  /** The id of the MultiSelect. */
+
+export interface MultiSelectProps {
+  /** The button text rendered within the MultiSelect. */
+  buttonText: string;
+  /** The number of items that will be visible in the list when the component first loads. */
+  defaultItemsVisible?: number;
+  /** The action to perform for clear/reset button of MultiSelect. */
+  onClear?: () => void;
+  /** The action to perform on the checkbox's onChange function. */
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** The action to perform for a mixed state checkbox (parent checkbox). */
+  onMixedStateChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** An ID string that other components can cross reference for accessibility purposes. */
   id: string;
+  /** Boolean value used to control how the MultiSelect component will render within the page
+   * and interact with other DOM elements.
+   * The default value is false. */
+  isBlockElement?: boolean;
   /** Set the default open or closed state of the Multiselect. */
   isDefaultOpen?: boolean;
-  /** Boolean value used to control how the MultiSelect component will render within the page and interact with other DOM elements.
+  /** Boolean value used to enable the component's search functionality
    * The default value is false. */
-  isBlockElement?: boolean;
+  isSearchable?: boolean;
   /** The items to be rendered in the Multiselect as checkbox options. */
   items: MultiSelectItem[];
-  /** The label text rendered within the MultiSelect. */
-  labelText: string;
-  /** The action to perform for clear/reset button of MultiSelect. */
-  onClear?: () => void;
-  /** The type of MultiSelect that will be rendered. */
-  type: "listbox" | "dialog";
+  /** listOverflow is a property indicating how the list should handle overflow, with options limited to either "scroll" or "expand." */
+  listOverflow?: MultiSelectListOverflowTypes;
   /** The selected items state (items that were checked by user). */
   selectedItems: SelectedItems;
   /** Value used to set the width for the MultiSelect component. */
   width?: MultiSelectWidths;
```

Comment:
I think we should stick with `width` for now. I would expect this to work because the component has `shouldForwardProp` set to `true`. If this is only in Storybook, then the comment can go directly in the `.stories.tsx` file (as tedious as it is).